packages/components/src/mobile/bottom-sheet-select-control/index.native.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * External dependencies
+ */
+import { SafeAreaView, View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { useNavigation } from '@react-navigation/native';
+import { useState } from '@wordpress/element';
+import { Icon, chevronRight, check } from '@wordpress/icons';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Cell from '../bottom-sheet/cell';
+import NavigationHeader from '../bottom-sheet/navigation-header';
+import SubSheet from '../bottom-sheet/sub-sheet';
+
+const BottomSheetSelectControl = ( {
+	label,
+	options: items,
+	onChange,
+	value: selectedValue,
+} ) => {
+	const [ showSubSheet, setShowSubSheet ] = useState( false );
+	const navigation = useNavigation();
+
+	const onChangeValue = ( value ) => {
+		return () => {
+			goBack();
+			onChange( value );
+		};
+	};
+
+	const selectedOption = items.find(
+		( option ) => option.value === selectedValue
+	);
+
+	const goBack = () => {
+		setShowSubSheet( false );
+		navigation.goBack();
+	};
+
+	const openSubSheet = () => {
+		navigation.navigate( SubSheet.screenName );
+		setShowSubSheet( true );
+	};
+
+	return (
+		<SubSheet
+			navigationButton={
+				<Cell
+					label={ label }
+					separatorType="none"
+					value={ selectedOption.label }
+					onPress={ openSubSheet }
+					accessibilityRole={ 'button' }
+					accessibilityLabel={ selectedOption.label }
+					accessibilityHint={ sprintf(
+						// translators: %s: Select control button label e.g. "Button width"
+						__( 'Navigates to select %s' ),
+						selectedOption.label
+					) }
+				>
+					<Icon icon={ chevronRight }></Icon>
+				</Cell>
+			}
+			showSheet={ showSubSheet }
+		>
+			<SafeAreaView>
+				<NavigationHeader
+					screen={ label }
```

Comment:
Good idea. I think we should tackle this in a different PR mostly since this one is already getting pretty big and it will require changing a few different components.